llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp

```diff
@@ -3239,50 +3239,76 @@ static SDValue performBitcastCombine(SDNode *N,
   return SDValue();
 }
 
-static SDValue performSETCCCombine(SDNode *N,
-                                   TargetLowering::DAGCombinerInfo &DCI) {
-  auto &DAG = DCI.DAG;
+template <int MatchRHS, ISD::CondCode MatchCond, bool RequiresNegate,
```

Comment:
Is there a particular reason for templating here as opposed to just passing in arguments? This will instantiate 4 different versions of the function which seems like a lot